db/users.go

```diff
@@ -178,3 +182,26 @@ func (ms *MongoStorage) IsMemberOf(userEmail, organizationAddress string, role U
 	}
 	return false, ErrNotFound
 }
+
+// UserByVerificationCode method returns the user with the given verification
+// code. If the user or the verification code doesn't exist, it returns a
+// specific error. If other errors occur, it returns the error. It checks the
+// user verification code in the verifications collection and returns the user
+// with the ID associated with the verification code.
+func (ms *MongoStorage) UserByVerificationCode(code string, t CodeType) (*User, error) {
+	ms.keysLock.RLock()
+	defer ms.keysLock.RUnlock()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	result := ms.verifications.FindOne(ctx, bson.M{"code": code, "type": t})
+	verification := &UserVerification{}
+	if err := result.Decode(verification); err != nil {
+		if err == mongo.ErrNoDocuments {
+			return nil, ErrNotFound
+		}
+		return nil, err
+	}
+	return ms.user(verification.ID)
```

Comment:
@lucasmenendez would there be an issue if you would use here the `User`? If that is the case, and together with the verifications comment above, no locks are needed and maybe the `user` helper function is not necessary anymore. What do you think?

Reply:
Probably i am not understand your comment... event this method is moved to `db/users.go`, the helper still being required to avoid duplicated code. The helper is private to ensure that it is used under a lock.